jerry-core/parser/js/opcodes-dumper.cpp

```diff
@@ -65,6 +65,484 @@ static vm_idx_t jsp_reg_max_for_args;
+{
+  jsp_operand_t ret;
+
+  ret.type = JSP_OPERAND_TYPE_SIMPLE_VALUE;
+  ret.data.simple_value = simple_value;
+
+  return ret;
+} /* jsp_make_simple_value_operand */
+
+/**
+ * Construct string literal operand
+ *
+ * @return constructed operand
+ */
+jsp_operand_t
+jsp_make_string_lit_operand (lit_cpointer_t lit_id) /**< literal identifier */
+{
+  JERRY_ASSERT (lit_id.packed_value != NOT_A_LITERAL.packed_value);
+
+#ifndef JERRY_NDEBUG
+  lit_literal_t lit = lit_get_literal_by_cp (lit_id);
+
+  JERRY_ASSERT (RCS_RECORD_IS_CHARSET (lit)
+                || RCS_RECORD_IS_MAGIC_STR (lit)
+                || RCS_RECORD_IS_MAGIC_STR_EX (lit));
+#endif /* !JERRY_NDEBUG */
+
+  jsp_operand_t ret;
+
+  ret.type = JSP_OPERAND_TYPE_STRING_LITERAL;
+  ret.data.lit_id = lit_id;
+
+  return ret;
+} /* jsp_make_string_lit_operand */
+
+/**
+ * Construct RegExp literal operand
+ *
+ * @return constructed operand
+ */
+jsp_operand_t
+jsp_make_regexp_lit_operand (lit_cpointer_t lit_id) /**< literal identifier */
+{
+  JERRY_ASSERT (lit_id.packed_value != NOT_A_LITERAL.packed_value);
+
+#ifndef JERRY_NDEBUG
+  lit_literal_t lit = lit_get_literal_by_cp (lit_id);
+
+  JERRY_ASSERT (RCS_RECORD_IS_CHARSET (lit)
+                || RCS_RECORD_IS_MAGIC_STR (lit)
+                || RCS_RECORD_IS_MAGIC_STR_EX (lit));
+#endif /* !JERRY_NDEBUG */
+
+  jsp_operand_t ret;
+
+  ret.type = JSP_OPERAND_TYPE_REGEXP_LITERAL;
+  ret.data.lit_id = lit_id;
+
+  return ret;
+} /* jsp_make_regexp_lit_operand */
+
+/**
+ * Construct number literal operand
+ *
+ * @return constructed operand
+ */
+jsp_operand_t
+jsp_make_number_lit_operand (lit_cpointer_t lit_id) /**< literal identifier */
+{
+  JERRY_ASSERT (lit_id.packed_value != NOT_A_LITERAL.packed_value);
+
+#ifndef JERRY_NDEBUG
+  lit_literal_t lit = lit_get_literal_by_cp (lit_id);
+
+  JERRY_ASSERT (RCS_RECORD_IS_NUMBER (lit));
+#endif /* !JERRY_NDEBUG */
+
+  jsp_operand_t ret;
+
+  ret.type = JSP_OPERAND_TYPE_NUMBER_LITERAL;
+  ret.data.lit_id = lit_id;
+
+  return ret;
+} /* jsp_make_number_lit_operand */
+
+/**
+ * Construct identifier reference operand
+ *
+ * @return constructed operand
+ */
+jsp_operand_t
+jsp_make_identifier_operand (lit_cpointer_t lit_id) /**< literal identifier */
+{
+  JERRY_ASSERT (lit_id.packed_value != NOT_A_LITERAL.packed_value);
+
+  jsp_operand_t ret;
+
+  ret.type = JSP_OPERAND_TYPE_IDENTIFIER;
+  ret.data.identifier = lit_id;
+
+  return ret;
+} /* jsp_make_identifier_operand */
+
+/**
+ * Construct register operand
+ *
+ * @return constructed operand
+ */
+jsp_operand_t
+jsp_make_reg_operand (vm_idx_t reg_index) /**< register index */
+{
+  /*
+   * The following check currently leads to 'comparison is always true
+   * due to limited range of data type' warning, so it is turned off.
+   *
+   * If VM_IDX_GENERAL_VALUE_FIRST is changed to value greater than 0,
+   * the check should be restored.
+   */
+  // JERRY_ASSERT (reg_index >= VM_IDX_GENERAL_VALUE_FIRST);
```

Comment:
I think its better to leave the assertion. There is a comment just above the assertion that describes why it is commented.